pkg/k8s/apis/cilium.io/v2alpha1/cnc_types.go

```diff
@@ -12,7 +12,6 @@ import (
 //+kubebuilder:resource:categories={cilium}
 //+kubebuilder:object:root=true
 //+deepequal-gen=false
-//+kubebuilder:storageversion
```

Comment:
In addition, could we use `+kubebuilder:unservedversion` to stop serving the old version anymore and rely on K8s to do the auto conversion over to the new version? That would allow us to git rid of the code marked by [1] and only change the code to assume `V2()` version. Meaning, we can rely on the k8s-apiserver to redirect requests for v2alpha1 to v2 and vice versa by virtue of versions being specified in the CRD spec. My assumption is that we don't need to explicitly keep serving the old version anymore, however I've not tried this.

Reply:
This would preclude downgrades, since the old version of cilium would be attempting to retrieve the v2alpha1 resource.

Reply:
> This would preclude downgrades, since the old version of cilium would be attempting to retrieve the v2alpha1 resource.

This is true only if the CRDs are also downgraded along with Cilium. However, I'm not sure if we are actually downgrading CRDs as well and whether we should.

cc @aanm

Reply:
Either way, up/downgrades are not atomic, so we don't know exactly what state the apiserver will have when starting Cilium.

If we were to remove `v2alpha1` from the served versions, we run the risk of a node starting up without the correct CiliumNodeConfig. That could lead to serious cluster breakage, depending on what is configured.

Reply:
But there are no timing guarantees between the operator and the agent. The operator could take 100 minutes to pull from Quay, in which case the agents will all be already updated and ready to run.